### ISSUE TITLE
Add --max-warnings <number> flag

### ISSUE THREAD

**Issue by revyh** [stylelint/stylelint#2939]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Feature request. I need to run stylelint on ci. And I want ci to report that checks fail if there are any stylelint warnings. To do so, I should be able to force stylelint to exit with non-zero exit code on warnings. I think that some cli flags would be enough to do the job. For example, I use [--max-warnings 0](https://github.com/eslint/eslint/issues/2769) to do same thing in eslint.

> Which rule, if any, is this issue related to?

none

> What CSS is needed to reproduce this issue?

```css
.foo {
  color: #color;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "color-no-invalid-hex": [true, {"severity": "warning"}]
  }
}
```

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint foo.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

stylint exits with non-zero exit code

> What actually happened (e.g. what warnings or errors you are getting)?

stylelint exits with zero code

It seems like it is a standard use case and it should be already reported before. But I can't find anything about it. So, sorry if this is a duplicate

**Comment by m-allanson** [comment]
Can you use `{"severity": "error"}` on CI? That should cause stylelint to exit with a non-zero code when it finds problems.

**Comment by jeddy3** [comment]
Yep, using a different config on the CI can be used to achieve this. However, I think the use case can be better handled via a `max-warnings` flag. In fact, there is already a pending issue for this feature: https://github.com/stylelint/stylelint/issues/2651. This new issue uses the issue template, though. So, I'll close the other issue in favour of this one. @revyh This issue, like the previous one, will be labelled as "help wanted". Please consider contributing this flag if you'd like to see it in stylelint.

**Comment by alexilyaev** [comment]
PR'ed #2942

**Comment by revyh** [comment]
@alexilyaev thx
